[GH1] Support merging PRs with more than 100 files

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73317
* #73316

Implement pagination in `GitHubPR.get_changed_files()` and add test for
that

malfet committed:
[GH1] Support merging PRs with more than 100 files

Implement pagination in `GitHubPR.get_changed_files()` and add test for
that

[ghstack-poisoned]

malfet committed:
Update on "[GH1] Support merging PRs with more than 100 files"


Implement pagination in `GitHubPR.get_changed_files()` and add test for
that

[ghstack-poisoned]

dreiss commented on .github/scripts/trymerge.py:
```diff
@@ -198,10 +221,20 @@ def get_changed_files_count(self) -> int:
+            info = self.info
+            self.changed_files = []
+            # Do not try to fetch more than 10K files
+            for _ in range(100):
+                self.changed_files += [x["path"] for x in info["files"]["nodes"]]
```
Maybe throw if this limit is hit?

```
while True:
  if len(self.changed_files > 10000):
    raise Whatever
```

malfet commented on .github/scripts/trymerge.py:
```diff
@@ -198,10 +221,20 @@ def get_changed_files_count(self) -> int:
+                self.changed_files += [x["path"] for x in info["files"]["nodes"]]
+                if not info["files"]["pageInfo"]["hasNextPage"]:
+                    break
+                rc = gh_graphql(GH_GET_PR_NEXT_FILES_QUERY, name=self.project, owner=self.org, number=self.pr_num, cursor=info["files"]["pageInfo"]["endCursor"])
```
Good point, updated

malfet committed:
Update on "[GH1] Support merging PRs with more than 100 files"


Implement pagination in `GitHubPR.get_changed_files()` and add test for
that

[ghstack-poisoned]

malfet commented:
@pytorchbot merge this
